### PR TITLE
write multiple files from plugin

### PR DESCRIPTION
**Pull request by TylerSehr**

### CODE REVIEW

**Comment by joeizang** [comment]
@TylerSehr Hi. Can you please describe the motivation behind the pr. It helps quite a bit.

File: apps/remix-ide/src/app/files/fileManager.ts [hunk 1]
```diff
@@ -217,6 +218,38 @@ class FileManager extends Plugin {
...
+  * @param {string[]} filePaths paths of the files
+  * @param {string[]} data content to write to each file
+  * @param {string} folderPath base folder path
+  * @returns {void}
+  */
+  async writeMultipleFiles(filePaths, fileData, folderPath) {
+    try {
+      let alert = true
+      for (let i = 0; i < filePaths.length; i++) {
+        let installPath = folderPath + "/" + filePaths[i]
```
Comment (joeizang) [inline apps/remix-ide/src/app/files/fileManager.ts:232]: Since installPath is not modified any where else. This can be a const rather than a let.

**Review by joeizang — CHANGES_REQUESTED** [review]

**Comment by TylerSehr** [comment]
writeMultipleFiles allows for a plugin to add entire protocols to the remix editor, preserve the file structure for the protocols, and only prompt once to write the files into a single folder. 

Currently plugins are able to do this using github gists, but that results in file collisions and certain files can be overwritten. 

We're updating the Cookbook plugin to support full protocols, so we want to be able to import an entire protocol into remix.

File: apps/remix-ide/src/app/files/fileManager.ts (hunk 1 above)
Comment (TylerSehr) [reply apps/remix-ide/src/app/files/fileManager.ts]: Updated installPath to const rather than let

**Review by joeizang — APPROVED** [review]
LGTM

File: apps/remix-ide/src/app/files/fileManager.ts [hunk 2]
```diff
@@ -217,6 +218,38 @@ class FileManager extends Plugin {
...
+          this.emit('fileAdded', path)
+        }
+        alert = false
+      }
+    } catch (e) {
+      throw new Error(e)
+    }
+  }
+
+
```
Comment (LianaHus) [inline apps/remix-ide/src/app/files/fileManager.ts:251]: Please remove extra lines here

**Comment by TylerSehr** [comment]
Added rename Cookbook Plugin

**Pull request merged**